python/ray/dag/compiled_dag_node.py

```diff
@@ -831,6 +829,19 @@ def _preprocess(self) -> None:
                         )
                     direct_input = True
 
+                elif isinstance(upstream_node.dag_node, ClassMethodNode):
+                    if (
+                        downstream_actor_handle is not None
+                        and downstream_actor_handle
+                        == upstream_node.dag_node._get_actor_handle()
+                        and upstream_node.dag_node.type_hint.requires_nccl()
+                    ):
+                        raise ValueError(
+                            "Compiled DAG does not support NCCL communication between "
```

Comment:
I think the msg should be more actionable. We should include.

1. what are 2 actors that have issues?
2. The name of upstream / downstream tasks (and ID). 

And how would we support this case?

Reply:
Done.